python_modules/dagster/dagster/_core/definitions/asset_layer.py

```diff
@@ -77,25 +77,26 @@ def from_graph_and_assets_node_mapping(
 
             for output_name, asset_key in assets_def.node_keys_by_output_name.items():
                 # resolve graph output to the op output it comes from
-                inner_output_def, inner_node_handle = (
-                    computation.full_node_def.resolve_output_to_origin(
-                        output_name, handle=node_handle
-                    )
+                full_node_def = computation.full_node_def
+                inner_output_def, inner_node_handle = full_node_def.resolve_output_to_origin(
+                    output_name, handle=node_handle
                 )
                 node_output_handle = NodeOutputHandle(
                     node_handle=inner_node_handle, output_name=inner_output_def.name
                 )
 
                 asset_keys_by_node_output_handle[node_output_handle] = asset_key
 
-                asset_key_by_input.update(
-                    {
-                        input_handle: asset_key
-                        for input_handle in computation.full_node_def.resolve_output_to_destinations(
-                            output_name, node_handle
-                        )
-                    }
+                destinations = computation.full_node_def.resolve_output_to_destinations(
```

Comment:
nit: use `full_node_def` var that you added

Reply:
oops, thanks for catching